Add broker recipe for installing Enterprise Broker

wanelo-pair opened the pull request:
This is only tested on SmartOS, but can easily be extended with additional groups in value_for_platform.

Commit:
Add broker recipe for installing Enterprise Broker

postwait commented on recipes/broker.rb:
```diff
@@ -0,0 +1,38 @@
+)
+napp_conf = value_for_platform(
+    "smartos" => {
+        "default" => "/opt/napp/etc/httpd.conf"
```
Napp doesn't use /opt/napp/etc/httpd.conf anymore.

It is baked into the noitd setup.

sax replied:
Ah hah. Is this something that we can fix in the recipe, then update on an older broker, or should it be run from scratch on a new node?

Commit:
Remove explicit 'java' dependency from broker recipe

We don't want to require the 'java' cookbook on nodes that do not install
a broker, so it is recommended rather than depends in the metadata. Java
should be pre-installed before running the broker recipe.
